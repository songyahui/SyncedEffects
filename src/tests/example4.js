"use hopscript"

var hh = require("hiphop");

var prg = <hh.module A=${{accessibility: hh.IN}} T V>
  <hh.local S>
    <hh.loop>
      <hh.abort A>
	<hh.sequence>
	  <hh.emit S/>
	  <hh.if S>
	    <hh.emit T/>
	  </hh.if>
	  <hh.pause/>
	  <hh.emit V/>
	</hh.sequence>
      </hh.abort>
    </hh.loop>
  </hh.local>
</hh.module>

exports.prg = new hh.ReactiveMachine(prg, "example4");
