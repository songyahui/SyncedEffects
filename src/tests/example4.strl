module example4:

input A;
output T, V;

signal S in
       loop
       		abort
       				emit S;
       				present S then emit T end;
       				pause;
       				emit V;
       		when A
       end loop
end signal

end module