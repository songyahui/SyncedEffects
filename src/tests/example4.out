example4> ;
--- Output: T
example4> ;
--- Output: T V
example4> ;
--- Output: T V
example4> ;
--- Output: T V
example4> ;
--- Output: T V
example4> A;
--- Output: T
example4> ;
--- Output: T V
example4> ;
--- Output: T V
example4> ;
--- Output: T V
example4> ;
--- Output: T V
